### DISCUSSION TITLE
C# support in CodeWars

### DISCUSSION THREAD

**Discussion by worlebird** [codewars/codewars.com#2701] [Q&A] [answered]
I tried out my first kata in C#, and was stumped for a bit. When I tried my code, I got a syntax error, because "IsNullOrWhitespace is not found in string." The top of my code has a "using System;", and the IsNullOrWhitespace() extension method to the string object has existed in .Net for YEARS, certainly much farther back than the version 10 of C# that is supposedly supported here.
Any idea what's going on here? It seems like some basic stuff may not be properly supported by CodeWars. This definitely cost me several minutes of time, and several failed test runs (not the biggest problem, but still frustrating). It also makes me wonder what other common framework methods I am going to run into that are not supported.

**Comment by kazk** [comment, marked as answer]
> I got a syntax error, because "IsNullOrWhitespace is not found in string."

Typo? [String.IsNullOrWhite**S**pace](https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorwhitespace?view=net-6.0).

**Reply by worlebird** [reply]
Ah, crap. Of course - the "Space" is capitalized.
Boy, you forget just how much you depend on the IDE to help out with Intellisense there.

**Reply by worlebird** [reply]
Oh, and thank you for your response - I should have said that initially :-)

**Comment by kazk** [comment]
> It seems like some basic stuff may not be properly supported by CodeWars. It also makes me wonder what other common framework methods I am going to run into that are not supported.

We're using the official [.NET SDK 6.0 image](https://hub.docker.com/_/microsoft-dotnet-sdk/) with `/langversion:10.0`. I haven't heard anything is missing.

**Reply by worlebird** [reply]
Yep - as you point out in your earlier response, a typo.
I never paid close attention to the capitalization of that one, since Intellisense fills it in for you before you finish it in Visual Studio.
It looked so obvious there on the screen.
